consensus_test.go

```diff
@@ -291,6 +303,75 @@ func TestTransition_AcceptState_Validator_LockCorrect(t *testing.T) {
 	})
 }
 
+func TestTransition_AcceptState_Validate_ProposalFail(t *testing.T) {
+	validateProposalFunc := func(proposal []byte) error {
+		return errors.New("failed to validate a proposal")
+	}
+
+	validatorIds := []string{"A", "B", "C"}
+	backend := newMockBackend(validatorIds, nil, nil, validateProposalFunc, nil)
+
+	m := newMockPbft(t, validatorIds, "C", backend)
+	m.state.view = ViewMsg(1, 0)
+	m.setState(AcceptState)
+
+	m.setProposal(&Proposal{
+		Data: mockProposal,
+		Time: time.Now(),
+	})
+
+	var buf bytes.Buffer
+	m.logger.SetOutput(&buf)
+	defer func() {
+		m.logger.SetOutput(getDefaultLoggerOutput())
+	}()
+
+	// Prepare messages
+	m.emitMsg(&MessageReq{
+		From: "A",
+		Type: MessageReq_Preprepare,
+		View: ViewMsg(1, 0),
+	})
+	m.emitMsg(&MessageReq{
+		From: "B",
+		Type: MessageReq_Preprepare,
+		View: ViewMsg(1, 0),
+	})
+	m.emitMsg(&MessageReq{
+		From: "C",
+		Type: MessageReq_Preprepare,
+		View: ViewMsg(1, 0),
+	})
+
+	go func() {
+		for {
+			if m.GetState() == RoundChangeState {
+				m.cancelFn()
+				return
+			}
+		}
+	}()
+
+	m.Run(m.ctx)
```

Comment:
Same comment as before for the `Run` function and the logger.

Reply:
What about this?

Reply:
Sorry, I have overseen this one.